SimPEG/EM/Static/DC/ProblemDC_2D.py

```diff
@@ -230,68 +341,71 @@ def getRHSDeriv(self, ky, src, v, adjoint=False):
         # return qDeriv
         return Zero()
 
-    def setBC(self):
-        if self.mesh.dim == 3:
-            fxm, fxp, fym, fyp, fzm, fzp = self.mesh.faceBoundaryInd
-            gBFxm = self.mesh.gridFx[fxm, :]
-            gBFxp = self.mesh.gridFx[fxp, :]
-            gBFym = self.mesh.gridFy[fym, :]
-            gBFyp = self.mesh.gridFy[fyp, :]
-            gBFzm = self.mesh.gridFz[fzm, :]
-            gBFzp = self.mesh.gridFz[fzp, :]
-
-            # Setup Mixed B.C (alpha, beta, gamma)
-            temp_xm = np.ones_like(gBFxm[:, 0])
-            temp_xp = np.ones_like(gBFxp[:, 0])
-            temp_ym = np.ones_like(gBFym[:, 1])
-            temp_yp = np.ones_like(gBFyp[:, 1])
-            temp_zm = np.ones_like(gBFzm[:, 2])
-            temp_zp = np.ones_like(gBFzp[:, 2])
+    def setBC(self, ky=None):
```

Comment:
In future, it would be interesting to see how much of the boundary condition code can be pulled out of this and into discretize itself.

Any thoughts on that @sgkang?

Reply:
Yeah. It needs to be. Too much redundant function ... I'll try to pull these into discretize.